kroxylicious-kms/src/main/java/io/kroxylicious/kms/service/Kms.java

```diff
@@ -70,5 +69,7 @@ public interface Kms<K, E> {
      * @throws KmsException For other exceptions.
      */
     @NonNull
-    CompletionStage<K> resolveAlias(@NonNull String alias);
-}
+    CompletionStage<KekId> resolveAlias(@NonNull String alias);
+
+    KekId unencryptedKekId();
```

Comment:
It feels like the policy decision about what to encrypt has burrowed too far into the KMS api. The decision still resides in the Filter code but the KMS api now has this concept of `unencryptedKekId` which I don't think means anything to the `KMS`.

I prefer the other implentation where we resolve a KEK id and then wrap it in some other object like an EncryptionScheme or policy. I think the KMS should only deals in facts it knows, that it could or couldn't resolve an alias.

Reply:
Dropped this from this PR as we decided that it was better addressed via https://github.com/kroxylicious/kroxylicious/issues/811